doc: allow strings in StringDecoder methods

sno2 opened the pull request:
The `StringDecoder#end()` and `StringDecoder#write()` functions did not include the `string` type as one of the acceptable types for the buffer argument. However, the JSDoc in the codebase states that `string` is allowed. Found by @ThatOneBro

See the following lines of code to validate the JSDoc in the codebase:

- https://github.com/nodejs/node/blob/main/lib/string_decoder.js#L94
- https://github.com/nodejs/node/blob/main/lib/string_decoder.js#L116

<!--
Before submitting a pull request, please read
https://github.com/nodejs/node/blob/HEAD/CONTRIBUTING.md.

Commit message formatting guidelines:
https://github.com/nodejs/node/blob/HEAD/doc/contributing/pull-requests.md#commit-message-guidelines

For code changes:
1. Include tests for any bug fixes or new features.
2. Update documentation if relevant.
3. Ensure that `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

mscdex commented:
I think we should either leave it out or at the very least add to the function descriptions to say that string arguments will not be processed (e.g. not treated as a "binary" string) and in the case of `write()` will be returned as-is.

sno2 commented:
I removed the `string` from the `buffer` type signature and added a note. Although, I'm wondering if this should just be dropped as you were talking about as it's weird behavior.

addaleax commented:
I wouldn’t document this. Passing a string to these functions does not make sense and is almost always a bug, it’s just supported for legacy compatibility here.

sno2 commented:
This does not make sense to include after considering the previous points.
